packages/graphql-client/src/api-client-utilities/operation-types.ts

```diff
@@ -2,6 +2,7 @@ export type InputMaybe<_R = never> = never;
 
 export interface AllOperations {
   [key: string]: { variables: any; return: any };
+  [key: number | symbol]: never;
```

Comment:
So TS is a bit weird with `[key: string]` - if you set it like that it will still allow the key type to be `number | symbol`, which I think is just because it internally knows how to convert those. We're just explicitly telling it MUST be a string so that we don't have to cast it.

Here is where that change came from: https://github.com/Shopify/shopify-app-js/pull/518#discussion_r1419157371